hydromt/model/components/grid.py

```diff
@@ -106,22 +104,38 @@ def set(
         if name is None and name_required:
             raise ValueError(f"Unable to set {type(data).__name__} data without a name")
         if isinstance(data, np.ndarray):
+            if data.size == 0:
+                raise ValueError("Cannot set grid data with empty array")
             if data.shape != self._data.raster.shape:
                 raise ValueError("Shape of data and grid maps do not match")
             data = xr.DataArray(dims=self._data.raster.dims, data=data, name=name)
-        elif isinstance(data, xr.DataArray):
+
+        if isinstance(data, xr.DataArray):
             if name is not None:
                 data.name = name
             data = data.to_dataset()
-        elif not isinstance(data, xr.Dataset):
+
+        if not isinstance(data, xr.Dataset):
             raise ValueError(f"cannot set data of type {type(data).__name__}")
 
+        # Check if noth is really up and south therefore is down
+        if data.raster.res[1] > 0:
+            data = data.raster.flipud()
```

Comment:
This is Wflow specific, e.g. SFINCS does not want this. Either remove or add an argument to the function for NS or SN orientation.